documentation/modules/deploying/proc-deploy-kafka-node-pools.adoc

```diff
@@ -0,0 +1,101 @@
+
+The deployment uses a YAML file to provide the specification to create a `KafkaNodePool` resource.
+You can use node pools with Kafka clusters that use KRaft (Kafka Raft metadata) mode or ZooKeeper for cluster management.
+
+IMPORTANT: **KRaft mode is not ready for production in Apache Kafka or in Strimzi.**
+
+Strimzi provides the following xref:config-examples-{context}[example files] that you can use to create a Kafka node pool:
+
+`kafka-with-dual-role-kraft-nodes.yaml`:: Deploys a Kafka cluster with one pool of KRaft nodes that share the broker and controller roles.
+`kafka-with-kraft.yaml`:: Deploys a Kafka cluster with one pool of controller nodes and one pool of broker nodes.
+`kafka.yaml`:: Deploys ZooKeeper with 3 nodes, and 2 different pools of Kafka brokers. Each of the pools has 3 brokers. The pools in the example use different storage configuration.
+
+NOTE: You don't need to start using node pools right away. If you decide to use them, you can perform the steps outlined here to deploy a new Kafka cluster with `KafkaNodePool` resources or xref:proc-migrating-clusters-node-pools-{context}[migrate your existing Kafka cluster].  
+
+.Prerequisites
+
+* xref:deploying-cluster-operator-str[The Cluster Operator must be deployed.]
+* You have xref:deploying-kafka-cluster-{context}[created and deployed a Kafka cluster]. 
+
+NOTE: If you want to migrate an existing Kafka cluster to use node pools, see the xref:proc-migrating-clusters-node-pools-{context}[steps to  migrate existing Kafka clusters].  
+
+.Procedure
+
+. Enable the `KafkaNodePools` feature gate.
++
+If using KRaft mode, enable the `UseKRaft` feature gate too.
++
+[source,yaml]
+----
+kubectl set env install/cluster-operator STRIMZI_FEATURE_GATES=""
+env:
+  - name: STRIMZI_FEATURE_GATES
+    value: +KafkaNodePools, +UseKRaft
+----
++
+This updates the Cluster Operator.
+
+. Create a node pool.
++
+* To deploy a Kafka cluster in KRaft mode with a  single node pool that uses dual-role nodes: 
++
+[source,shell,subs="attributes+"]
+kubectl apply -f examples/kafka/nodepools/kafka-with-dual-role-kraft-nodes.yaml
+
+* To deploy a Kafka cluster in KRaft mode with separate node pools for broker and controller nodes: 
++
+[source,shell,subs="attributes+"]
+kubectl apply -f examples/kafka/nodepools/kafka-with-kraft.yaml
+
+* To deploy a Kafka cluster and ZooKeeper cluster with two node pools of three brokers: 
++
+[source,shell,subs="attributes+"]
+kubectl apply -f examples/kafka/nodepools/kafka.yaml
+
+. Check the status of the deployment:
++
+[source,shell,subs="+quotes"]
+----
+kubectl get pods -n _<my_cluster_operator_namespace>_
+----
++
+.Output shows the node pool names and readiness
+[source,shell,subs="+quotes"]
+----
+NAME                        READY   STATUS    RESTARTS
+my-cluster-entity-operator  3/3     Running   0
+my-cluster-pool-a-kafka-0   1/1     Running   0
+my-cluster-pool-a-kafka-1   1/1     Running   0
+my-cluster-pool-a-kafka-4   1/1     Running   0
```

Comment:
Is it really worth showing the output? I mean ... previously you are explaining there are 3 different examples of using node pools but it's not clear to which this output is referring to. We could be more precise and clear or just remove this output?

Reply:
We're showing here a couple of things: the additional node is added and running and also we have the note to say that the node IDs might not be sequential. I think this is a useful verification that could anyway be skipped.

Reply:
So .... you are putting this output here after mentioning the above three examples we are providing so I think that should be some relationship between this output and the above examples but ...

- `pool-a` is the name of the pool for the kafka.yaml example only. All the other examples have different pool names.
- deploying the kafka.yaml example you will have 0,1 and 2 as IDs ... not 4

What I mean, if the user just apply the above YAMLs file as you are describing and compare with the output you are showing, they won't match. It could be confusing imho.

Reply:
please see https://github.com/strimzi/strimzi-kafka-operator/pull/8683#issuecomment-1601166058